client/app/src/services/network.service.js

```diff
@@ -347,22 +351,65 @@
 
     // Updates peer with all currency values relative to the USD price.
     function updatePeerWithCurrencies (peer, res) {
-      $http.get('https://api.fixer.io/latest?base=USD', {timeout: 2000}).then(function (result) {
-        const USD_PRICE = Number(res.data[0].price_usd)
-        var currencies = ['aud', 'brl', 'cad', 'chf', 'cny', 'eur', 'gbp', 'hkd', 'idr', 'inr', 'jpy', 'krw', 'mxn', 'rub']
-        var prices = {}
-        currencies.forEach(function (currency) {
-          prices[currency] = result.data.rates[currency.toUpperCase()] * USD_PRICE
-        })
-        prices['btc'] = res.data[0].price_btc
-        prices['usd'] = res.data[0].price_usd
-        peer.market.price = prices
-        storageService.setGlobal('peerCurrencies', prices)
+      peer = updateCurrencyConversionRates(peer)
+      const USD_PRICE = Number(res.data[0].price_usd)
+      var currencies = ['AUD', 'BRL', 'CAD', 'CHF', 'CNY', 'EUR', 'GBP', 'HKD', 'IDR', 'INR', 'JPY', 'KRW', 'MXN', 'RUB']
+      var prices = {}
+      currencies.forEach(function (currency) {
+        prices[currency.toLowerCase()] = peer.market.conversionRates[currency] * USD_PRICE
       })
+      prices['btc'] = res.data[0].price_btc
+      prices['usd'] = res.data[0].price_usd
+      storageService.setGlobal('peerCurrencies', prices)
+      peer.market.price = prices
+      return peer
+    }
 
+    // Updates the currency conversion rates IF necessary
+    // Necessary if it isn't stored, or if the stored value is too old
+    function updateCurrencyConversionRates (peer) {
+      var priceObj = storageService.getGlobal('conversionRates')
+      if (priceObj !== undefined) {
```

Comment:
I think you will need to check for `!== null` here too. When you reset all data in the client, priceObj is null, which !== undefined.

Also, when i was force overriding priceObj.date to a few days ago, updateCurrencies was still being set to false, so it wasn't updating the price. From what I can tell, this results in the price never being updated again. My code from debugging in case it helps:

```
    // Updates the currency conversion rates IF necessary
    // Necessary if it isn't stored, or if the stored value is too old
    function updateCurrencyConversionRates (peer) {
      var priceObj = storageService.getGlobal('conversionRates')
      console.log('priceObj', priceObj)
      if (priceObj !== undefined && priceObj !== null) {
        priceObj.date = "2017-12-26T16:19:29.679Z"
        peer.market.conversionRates = priceObj.rates
        let storedDateString = priceObj.date
        let storedDate = new Date(storedDateString)
        console.log('storedDate', storedDate)
        console.log('priceObj', priceObj)
        var updateCurrencies = checkToUpdateConversionRates(storedDate)
        console.log('updateCurrencies', updateCurrencies)
        if (updateCurrencies) {
          getConversionRatesApiCall(peer)
        }
      } else {
        getConversionRatesApiCall(peer)
      }
      return peer
    }

    // api call to get the conversion rates for currencies
    function getConversionRatesApiCall (peer) {
      console.log('getConversionRatesApiCall', peer)
      var currencies = ['AUD', 'BRL', 'CAD', 'CHF', 'CNY', 'EUR', 'GBP', 'HKD', 'IDR', 'INR', 'JPY', 'KRW', 'MXN', 'RUB']
      var apiCall = createCurrencyConversionApiCall(currencies)
      $http.get(apiCall, {timeout: 2000}).then(function (result) {
        storageService.setGlobal('conversionRates', { rates: result.data.rates, date: new Date() })
        peer.market.conversionRates = result.data.rates
      })
      return peer
    }
```

Could you test and see if you can get it to force update the price (and make sure I'm not being thick)

Reply:
Nah, I figured out my logic error. I was the dunce :(

Basically I had it set so that the 'update time' is the same day as the storedDate, but if hte stored date is past the 4PM CET then it won't pass the range test, because the update time is before the storedDate.

I got it fixed by checking to see if it's after 4pm (hour 16), if so then moment lets you add a day, I'll have that up in a bit with some more testing.

Good catch.